tests/contrib/celery/test_integration.py

```diff
@@ -0,0 +1,225 @@
+from nose.tools import eq_, ok_
+
+from .utils import CeleryTestCase
+
+
+class CeleryIntegrationTask(CeleryTestCase):
+    """
+    Ensures that the tracer works properly with a real Celery application
+    without breaking the Application or Task APIs.
+    """
+    def test_concurrent_delays(self):
+        # it should create one trace for each delayed execution
+        @self.app.task
+        def fn_task():
+            return 42
+
+        for x in range(100):
+            fn_task.delay()
+
+        traces = self.tracer.writer.pop_traces()
```

Comment:
So with this patching we trace two parts of `celery`, the code which schedules a task to be executed and the execution of the task.

In this test we are using `.delay()` which is a task scheduler, so the function will block until the task is pushed into our broker. It won't wait until the task has completed, and that is totally fine/expected.